Multiple files: cleanup common include files

mykaul opened the pull request:
As a continuation to #3130 , there are several .h files that are included in various files.
However, many of the definitions in those include files should be local to specific files, or is simply dead code.
We can easily clean them up.
Note - there are some functions that were deemed useful as common or utilities that were meant to be shared across the codebase.
After 10 or so years, I think it's OK to move some of them that were never shared across the code,
to their own users (and make them static while at it).

Updates: #3137
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
Multiple files: cleanup common include files

As a continuation to #3130 , there are several .h files that are included in various files.
However, many of the definitions in those include files should be local to specific files, or is simply dead code.
We can easily clean them up.
Note - there are some functions that were deemed useful as common or utilities that were meant to be shared across the codebase.
After 10 or so years, I think it's OK to move some of them that were never shared across the code,
to their own users (and make them static while at it).

Updates: #3137
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/distribute/spare_file_rebalance.t

0 test(s) generated core 


8 test(s) needed retry 
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/basic/afr/sparse-file-self-heal.t
./tests/basic/distribute/spare_file_rebalance.t

5 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2049/

mykaul commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index b2dbde70c..dab8158b1 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -298,7 +298,8 @@ gf_gfid_generate_from_xxh64(uuid_t gfid, char *key)
     }
 
     gf_msg_debug(this->name, 0,
-                 "gfid generated is %s (hash1: %" XXH64_FMT ") "
+                 "gfid generated is %s (hash1: %" XXH64_FMT
+                 ") "
                  "hash2: %" XXH64_FMT ", xxh64_1: %s xxh64_2: %s",
                  uuid_utoa(gfid), hash_1, hash_2, xxh64_1, xxh64_2);
 
```

mykaul commented:
/run regression

mykaul commented:
/run regression

amarts commented:
conflicts :-/

Commit:
Multiple files: cleanup common include files - mainly glusterd-utils.c

As a continuation to #3130 , there are several .h files that are included in various files.
However, many of the definitions in those include files should be local to specific files, or is simply dead code.
We can easily clean them up.
Note - there are some functions that were deemed useful as common or utilities that were meant to be shared across the codebase.
After 10 or so years, I think it's OK to move some of them that were never shared across the code,
to their own users (and make them static while at it).

This commit specifically cleaned up glusterd-utils.c

    Updates: #3137
    Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression
